xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/notifications/sources/internal/LegacyEventSearcher.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.sources.internal;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.eventstream.Event;
+import org.xwiki.eventstream.EventStream;
+import org.xwiki.eventstream.EventStreamException;
+import org.xwiki.notifications.sources.NotificationParameters;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+
+/**
+ * Search {@link Event}.
+ * 
+ * @version $Id$
+ * @since 14.6RC1
+ */
+@Component(roles = EventSearcher.class)
```

Comment:
It means it overrides the default EventSearcher component in the CM? There couldn't be any problem of which one is loaded first?

Reply:
There are two possibilities:
* it's a core extension (the current situation): there is an override in the component.txt
* it's an installed extension: it's always registered after the other one, so the legacy one wins